medfile 4.1.0 fails to build with HDF5 version 1.12.0

**Describe the bug**
The package medfile-4.1.0 fails to build on x86_64 with the following error:
```
HDF5 version is 1.12.0.  Only versions >= 1.10.2 are supported.
```

**To Reproduce**
Steps to reproduce the behavior:
1. Switch to `nixos-unstable-small` (764448055551a4c664a1b0e2cecfc61731b32c5d)
2. Run `nix-shell -p medfile`
3. Build fails:
```
-- Check for HDF5 ...
-- Found HDF5: /nix/store/n6gic5akqirw4bfghnlmzqa5fyylhr5g-hdf5-1.12.0/lib/libhdf5.so;/nix/sto>
-- Found HDF5 version 1.12.0 -- parallel flag is set to 
-- _versionComponents=1;12;0
-- HDF5_VERSION=1.12.0
CMake Error at config/cmake_files/medMacros.cmake:451 (MESSAGE):
  HDF5 version is 1.12.0.  Only versions >= 1.10.2 are supported.
Call Stack (most recent call first):
  CMakeLists.txt:110 (MED_FIND_HDF5)
  ```

**Expected behavior**
Medfile should be able to build

**Additional context**
I guess the version check will be just HDF5_VERSION == 1.11.*, I will look into creating a patch. 

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
No maintainers, but @r-ryantm did the last meaningful change.

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.9, NixOS, 21.05.20210411.7644480 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210326_dd77f71`
 - channels(rien): `""`
 - nixpkgs: `/etc/nixpkgs`
 ```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6031,7 +6031,9 @@ in
     inherit (python3Packages) ansi2html;
   };
 
-  medfile = callPackage ../development/libraries/medfile { };
+  medfile = callPackage ../development/libraries/medfile {
+    hdf5 = hdf5_1_10;
+  };
 
   meilisearch = callPackage ../servers/search/meilisearch {
     inherit (darwin.apple_sdk.frameworks) IOKit Security;

